### PR TITLE
Add to linalg trace, zero, identity, add_col_vec (CPU only)

### PR DESCRIPTION
**Pull request by micmn**
Operations needed by #3826, linalg + eigen implementation.

~~This is a work in progress: tests, docs, dimension checks etc are missing.~~

PS: if I run `check_format.sh` it wants to rewrite almost completely `LinalgBackendEigen.h`, see https://gist.github.com/micmn/8565fdbb0bcb1e2b1867a934f8f0fbce#file-style-diff-L193

### CODE REVIEW

**Comment by vigsterkr** [comment]
@OXPHOS if you by any chance happen to have some time to give a glimpse and blessing on this that'd be appreciated!

**Comment by vigsterkr** [comment]
@micmn hehe yeah i reckon we can omit the style refactor in this case :P

**Comment by OXPHOS** [comment]
Cool I'll check ASAP!

File: src/shogun/mathematics/linalg/LinalgNamespace.h [hunk 1]
```diff
@@ -979,6 +979,94 @@ SGVector<T> rowwise_sum(const Block<SGMatrix<T>>& a, bool no_diag=false)
...
+ */
+template <typename T>
+T trace(const SGMatrix<T>& A)
+{
+	REQUIRE(A.num_rows == A.num_cols, "Matrix is not square!\n");
+	return infer_backend(A)->trace(A);
+}
+
+/**
+ * Method that write the identity into a square matrix.
```
Comment (OXPHOS) [inline src/shogun/mathematics/linalg/LinalgNamespace.h]: writes

File: src/shogun/mathematics/linalg/LinalgNamespace.h [hunk 2]
```diff
@@ -979,6 +979,94 @@ SGVector<T> rowwise_sum(const Block<SGMatrix<T>>& a, bool no_diag=false)
...
+ * Or pass the operand argument b as a result.
+ *
+ * @param A The matrix
+ * @param b The vector
+ * @param result The vector that saves the result
+ * @param alpha Constant to be multiplied by the matrix
+ * @param beta Constant to be multiplied by the vector
+ */
+template <typename T>
+void add_col_vec(
```
Comment (OXPHOS) [inline src/shogun/mathematics/linalg/LinalgNamespace.h]: Is this method used anywhere in Shogun? It is kinda specified

**Comment by OXPHOS** [comment]
I'm good with all the changes. I used to list the methods in alphabetic order for the convenience (to a certain degree). Can you adjust them if possible?  

Also @micmn you're working on algorithms cleanup project, right? If so, I won't ask you to try GPU backends. But you can specify in the title that these methods are only available on CPU.

**Comment by OXPHOS** [comment]
The format issue can be fixed later.

File: src/shogun/mathematics/linalg/LinalgNamespace.h (hunk 2 above)
Comment (OXPHOS) [inline src/shogun/mathematics/linalg/LinalgNamespace.h]: Oh never mind. I saw it.

**Comment by micmn** [comment]
>  I used to list the methods in alphabetic order for the convenience (to a certain degree)

Whoa! I feel stupid not to have noticed :)

> If so, I won't ask you to try GPU backends. But you can specify in the title that these methods are only available on CPU.
Yeah, currently I don't have ViennaCL setup in my system but it would be nice to have a try later.

**Pull request merged**